### PR TITLE
Email verification and change flows

### PR DESCRIPTION
**Pull request by pfrazee**
- Adds email verification flow
- Adds email change flow
- Adds a reminder on app load to verify email if not verified. Reminder only shows once a day and shards the users into 2 days of the week to get it, to avoid nagging too much and to avoid everybody running the flow all at once.

## Settings
CleanShot 2023-09-27 at 15 30 43@2x

## Verify flow

CleanShot 2023-09-27 at 15 30 52@2x
CleanShot 2023-09-27 at 15 31 04@2x

## Change flow

CleanShot 2023-09-27 at 15 31 43@2x
CleanShot 2023-09-27 at 15 31 57@2x
CleanShot 2023-09-27 at 15 32 16@2x

## Reminder on app open

CleanShot 2023-09-27 at 15 32 43@2x

### CODE REVIEW

File: src/state/models/ui/reminders.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import {toHashCode} from 'lib/strings/helpers'
+
+const DAY = 60e3 * 24 * 1 // 1 day (ms)
+
+export class Reminders {
+  // NOTE
+  // by defaulting to the current date, we ensure that the user won't be nagged
+  // on first run (aka right after creating an account)
+  // -prf
+  lastEmailConfirm: Date = new Date()
```
Comment (estrattonbailey) [inline src/state/models/ui/reminders.ts:13]: Nice

File: src/view/com/modals/ChangeEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                please verify your new email.
+              </>
+            )}
+          </Text>
+
+          {stage === Stages.InputEmail && (
+            <TextInput
+              testID="emailInput"
+              style={[styles.textInput, pal.border, pal.text]}
+              placeholder="alice@mail.com"
```
Comment (estrattonbailey) [inline src/view/com/modals/ChangeEmail.tsx:132]: Should pry disable autocomplete, autocapitalize, etc

**Review by estrattonbailey — APPROVED** [review]
LGTM

Nit on the auto* attributes.

Could probably clarify the error copy when trying to change your email prior to having first verified the current email. Rn it just says `confirmation token required`, but really it's `You must first verify your current email before changing it.`

File: src/view/com/modals/ChangeEmail.tsx (hunk 2 above)
Comment (pfrazee) [reply src/view/com/modals/ChangeEmail.tsx:140]: Done

**Comment by pfrazee** [comment]
> Could probably clarify the error copy when trying to change your email prior to having first verified the current email.

Done

**Pull request merged**